### PR TITLE
[DeadCode] Do not remove parameter on RemoveUnusedPrivatePropertyRector on constructor only usage in the middle of parameter

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
class SkipConstructorOnlyInMiddleParameterUsed
{
    /**
     * @var int
     */
    private $contentFinder;

    public function __construct(int $contentFinder, \stdClass $stdClass)
    {
        $this->contentFinder = $contentFinder;
        $this->init($stdClass);
    }

    private function init(\stdClass $stdClass)
    {
        var_dump($stdClass);
    }
}
```

It produce:

```diff
-    public function __construct(int $contentFinder, \stdClass $stdClass)
+    public function __construct(\stdClass $stdClass)
     {
-        $this->contentFinder = $contentFinder;
         $this->init($stdClass);
```

which can make the caller of the class error. It needs to be skipped.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
The constructor might be BC break in case of scalar values. But the removal of property assign and property seems legit.

**Comment by samsonasik** [comment]
This is only effect if the removed parameter not in the last used parameter, as the caller which call:

```php
(new SkipConstructorOnlyInMiddleParameterUsed(1, new stdClass));
```

will got error:

```bash
Uncaught TypeError: SkipConstructorOnlyInMiddleParameterUsed::__construct(): Argument #1 ($stdClass) must be of type stdClass, int given
```

ref https://3v4l.org/A6PvC

**Comment by samsonasik** [comment]
Ok, it seems the removal should be only for : 

- property
- assign

but keep the constructor parameter.

**Comment by samsonasik** [comment]
@TomasVotruba I updated failing fixture to only remove property and assign ,but keep the constructor parameter as it happen not in last parameter

https://github.com/rectorphp/rector-src/pull/1212/commits/1f72f8e98c1defeaa516aa09767bc0101f8c3968

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/DowngradePhp70/Rector/FuncCall/DowngradeDirnameLevelsRector/DowngradeDirnameLevelsRectorTest.php [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
-        return __DIR__ . '/config/configured_rule.php';
-    }
-}
\ No newline at end of file
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\DowngradePhp70\Rector\FuncCall\DowngradeDirnameLevelsRector;
+
```
Comment (TomasVotruba) [inline rules-tests/DowngradePhp70/Rector/FuncCall/DowngradeDirnameLevelsRector/DowngradeDirnameLevelsRectorTest.php]: Btw, what has changed here? The file looks identical. EOLs?

**Comment by TomasVotruba** [comment]
Thank you :clap:

File: rules-tests/DowngradePhp70/Rector/FuncCall/DowngradeDirnameLevelsRector/DowngradeDirnameLevelsRectorTest.php (hunk 1 above)
Comment (samsonasik) [reply rules-tests/DowngradePhp70/Rector/FuncCall/DowngradeDirnameLevelsRector/DowngradeDirnameLevelsRectorTest.php]: that's seems spacing and eol

**Pull request merged**